pipeline_tools/tests/test_confirm_submission.py

```diff
@@ -2,23 +2,27 @@
 import requests
 import requests_mock
 from pipeline_tools import confirm_submission
-from tenacity import stop_after_attempt, RetryError
+from pipeline_tools.http_requests import HttpRequests
+from .http_requests_manager import HttpRequestsManager
+from tenacity import RetryError
 
 
 class TestConfirmSubmission(unittest.TestCase):
 
     @requests_mock.mock()
-    def test_wait_for_valid_status(self, mock_request):
+    def test_wait_for_valid_status_success(self, mock_request):
         envelope_url = 'http://api.ingest.dev.data.humancellatlas.org/submissionEnvelopes/abcde'
 
         def _request_callback(request, context):
             context.status_code = 200
             return {'submissionState': 'Valid'}
 
         mock_request.get(envelope_url, json=_request_callback)
-        status = confirm_submission.wait_for_valid_status(envelope_url)
-        self.assertEqual(status, 'Valid')
-        self.assertEqual(mock_request.call_count, 1)
+
+        with HttpRequestsManager():
+            result = confirm_submission.wait_for_valid_status(envelope_url, HttpRequests())
+            self.assertEqual(result, True)
```

Comment:
I think these assert statements should be outside of the `with` block

Reply:
Oh actually I'm thinking about how assert statements inside of a `with self.assertRaises(requests.HTTPError)` block would not raise errors in the test. It may not matter in this case with the `HttpRequestsManager`